### ISSUE TITLE
App makes too much requests if open sitemap isn't found anymore

### ISSUE THREAD

**Issue by mueller-ma** [openhab/openhab-android#741]
### Actual behaviour
The app makes about 100 requests per second.

### Expected behaviour
An error should be shown instead.

 
### Steps to reproduce
1. Open a sitemap in the app
2. Remove sitemap file on the server
3. Manually reload in the app

### Can you reproduce the issue in demo mode?

I cannot change the sitemaps there, so no.

#### Client
* Android version: Android 7.1.2
* Device model: Oneplus X
* Stock or customized ROM: LineageOS 14.1
* App version : 2.0.15-beta
* Installation source : Github
* Device language: DE

#### Server
* Server version: 2.2
* Reverse Proxy: none
* Authentication method : none

#### App log
```
03-05 11:30:08.688 32350-32350/org.openhab.habdroid I/OpenHABWidgetListFragment:  showPage for http://openhab:8080/rest/sitemaps/demo2/demo2 longPolling = false
03-05 11:30:08.689 32350-32350/org.openhab.habdroid D/OpenHABWidgetListFragment: isAdded = true
03-05 11:30:08.689 32350-32350/org.openhab.habdroid D/OpenHABWidgetListFragment: Start progress indicator
03-05 11:30:08.705 32350-32350/org.openhab.habdroid D/OpenHABWidgetListFragment: Stop progress indicator
03-05 11:30:08.705 32350-32350/org.openhab.habdroid E/OpenHABWidgetListFragment: java.io.IOException: Not Found
03-05 11:30:08.705 32350-32350/org.openhab.habdroid E/OpenHABWidgetListFragment: status code = 404
03-05 11:30:08.705 32350-32350/org.openhab.habdroid E/OpenHABWidgetListFragment: Connection error = class java.io.IOException, cycle aborted

```

#### openHAB Server log
```
2018-03-05 11:27:47.627 [INFO ] [est.sitemap.internal.SitemapResource] - Received HTTP GET request at 'http://openhab:8080/rest/' for the unknown sitemap 'demo2'.
2018-03-05 11:27:47.639 [INFO ] [est.sitemap.internal.SitemapResource] - Received HTTP GET request at 'http://openhab:8080/rest/' for the unknown sitemap 'demo2'.
2018-03-05 11:27:47.656 [INFO ] [est.sitemap.internal.SitemapResource] - Received HTTP GET request at 'http://openhab:8080/rest/' for the unknown sitemap 'demo2'.
2018-03-05 11:27:47.673 [INFO ] [est.sitemap.internal.SitemapResource] - Received HTTP GET request at 'http://openhab:8080/rest/' for the unknown sitemap 'demo2'.
2018-03-05 11:27:47.692 [INFO ] [est.sitemap.internal.SitemapResource] - Received HTTP GET request at 'http://openhab:8080/rest/' for the unknown sitemap 'demo2'.
2018-03-05 11:27:47.710 [INFO ] [est.sitemap.internal.SitemapResource] - Received HTTP GET request at 'http://openhab:8080/rest/' for the unknown sitemap 'demo2'.
2018-03-05 11:27:47.742 [INFO ] [est.sitemap.internal.SitemapResource] - Received HTTP GET request at 'http://openhab:8080/rest/' for the unknown sitemap 'demo2'.
2018-03-05 11:27:47.761 [INFO ] [est.sitemap.internal.SitemapResource] - Received HTTP GET request at 'http://openhab:8080/rest/' for the unknown sitemap 'demo2'.
2018-03-05 11:27:47.778 [INFO ] [est.sitemap.internal.SitemapResource] - Received HTTP GET request at 'http://openhab:8080/rest/' for the unknown sitemap 'demo2'.
2018-03-05 11:27:47.794 [INFO ] [est.sitemap.internal.SitemapResource] - Received HTTP GET request at 'http://openhab:8080/rest/' for the unknown sitemap 'demo2'.
2018-03-05 11:27:47.809 [INFO ] [est.sitemap.internal.SitemapResource] - Received HTTP GET request at 'http://openhab:8080/rest/' for the unknown sitemap 'demo2'.
2018-03-05 11:27:47.825 [INFO ] [est.sitemap.internal.SitemapResource] - Received HTTP GET request at 'http://openhab:8080/rest/' for the unknown sitemap 'demo2'.
2018-03-05 11:27:47.842 [INFO ] [est.sitemap.internal.SitemapResource] - Received HTTP GET request at 'http://openhab:8080/rest/' for the unknown sitemap 'demo2'.
2018-03-05 11:27:47.859 [INFO ] [est.sitemap.internal.SitemapResource] - Received HTTP GET request at 'http://openhab:8080/rest/' for the unknown sitemap 'demo2'.
2018-03-05 11:27:47.876 [INFO ] [est.sitemap.internal.SitemapResource] - Received HTTP GET request at 'http://openhab:8080/rest/' for the unknown sitemap 'demo2'.
2018-03-05 11:27:47.892 [INFO ] [est.sitemap.internal.SitemapResource] - Received HTTP GET request at 'http://openhab:8080/rest/' for the unknown sitemap 'demo2'.
2018-03-05 11:27:47.910 [INFO ] [est.sitemap.internal.SitemapResource] - Received HTTP GET request at 'http://openhab:8080/rest/' for the unknown sitemap 'demo2'.

```

**Comment by maniac103** [comment]
Should fix this after #688 by notifying the activity from PageConnectionHoldePageConnectionHolderFragment after a 404 error, after which the activity triggers a sitemap reload.

**Comment by schinken** [comment]
Actually this also happens to me on not-found items. Record was 400 requests/second which was killing our openhab instance running into OutOfMemory
